### ISSUE TITLE
S3: custom s3 endpoint

### ISSUE THREAD

**Issue by l15k4** [akka/alpakka#181]
Hey,

I'm using https://github.com/jubos/fake-s3 extensively in many projects and I'm very satisfied with it so far. 
It requires the aws s3 client to be set with : 
```
s3.setEndpoint(s"http://localhost:1234")
s3.setS3ClientOptions(S3ClientOptions.builder().setPathStyleAccess(true).disableChunkedEncoding().build())
```
for it to work. At least the endpoint, the second setting is not necessary. Is there a way to change endpoint without changing code? It is not possible with the `proxy`

**Comment by francisdb** [comment]
Interesting, I wonder if there is a similar project for the jvm

**Comment by carstenlenz** [comment]
We use https://github.com/findify/s3mock 
Sadly there are some issues with Alpakka S3 and S3Mock.

**Comment by l15k4** [comment]
Yeah, I've been using both. But I'm way more productive with fake-s3, s3mock had too many issues, fake-s3 is almost flawless for testing purposes.

**Comment by ennru** [comment]
The work of #619 is now continued in #677

**Comment by easel** [comment]
For reference, there is a (passing) unit test in #619 that uses custom endpoints to connect to minio, including a one-liner to set up and configure minio for testing using docker. From what I can tell it's a fairly complete s3 implementation and worked as expected after I made a tweak to the url encoding of paths.

**Comment by ennru** [comment]
Fixed with #677
